pkg/controller/operators/catalog/operator.go

```diff
@@ -1336,6 +1362,35 @@ func (o *Operator) validateCustomResourceDefinition(oldCRD *v1beta1ext.CustomRes
 	return nil
 }
 
+// Validate all existing served versions against new CRD's validation (if changed)
+func (o *Operator) validateV1CustomResourceDefinition(oldCRD *v1ext.CustomResourceDefinition, newCRD *v1ext.CustomResourceDefinition) error {
+	o.logger.Debugf("Comparing %#v to %#v", oldCRD.Spec.Versions, newCRD.Spec.Versions)
+
+	// If validation schema is unchanged, return right away
+	for i := range oldCRD.Spec.Versions {
+		if i < len(newCRD.Spec.Versions) && reflect.DeepEqual(oldCRD.Spec.Versions[i].Schema, newCRD.Spec.Versions[i].Schema) {
```

Comment:
this is again trying to guard against index out of bounds panics if the version lengths between old and new CRDs are different